src/django/pfb_analysis/models.py

```diff
@@ -329,14 +337,25 @@ def create_environment(**kwargs):
         try:
             self.batch_job_id = response['jobId']
             self.save()
-            AnalysisJobStatusUpdate.objects.create(job=self,
-                                                   status=self.Status.QUEUED,
-                                                   step=self.Status.QUEUED)
+            self.update_status(self.status.QUEUED)
         except (botocore.exceptions.BotoCoreError, KeyError):
             logger.exception('Error starting AnalysisJob {}'.format(self.uuid))
 
+    def update_status(self, status, step='', message=''):
+        if self.status != self.Status.CANCELLED:
+            AnalysisJobStatusUpdate.objects.create(job=self,
```

Comment:
The same note applies the the way I originally wrote the status updating command, but it occurs to me now that this could be written as `self.status_updates.create(status=status, step=step, message=message)`, which is shorter and seems slightly more correct.